File: foodline.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

#define MAX_LINES 100

// Create a helper function to find the shortest line
int shortest_line_index(int lines[], int n) {
	int shortest = 0;
	for (int j = 1; j < n; ++j) {
		if (lines[j] < lines[shortest]) {
			shortest = j;
		}
	}
	return shortest;
}

void solve(int lines[], int n, int m) {
	int shortest;
	for (int i = 0; i < m; ++i) {
		shortest = shortest_line_index(lines, n);
		printf("%d\n", lines[shortest]);
		lines[shortest]++;
	}
}

int main() {
	int lines[MAX_LINES];
	int n, m, i;
	scanf("%d%d", &n, &m);
	for (i = 0; i < n; ++i) {
		scanf("%d", &lines[i]);
	}
	solve(lines, n, m);
	return 0;
}

File: Foodline/foodline.zig
const std = @import("std");

const MAX_LINES = 100;

fn shortestLineIndex(lines: []u32, n: usize) usize {
    var shortest: usize = 0;
    for (lines[1..n]) |line, j| {
	    if (line < lines[shortest]) {
		    shortest = j + 1;
		}
	}
    return shortest;
}

fn solve(lines: []u32, n: usize , m: usize) void {
    var stdout = std.io.getStdOut().writer();
    for (m) |i| {
	    const shortest = shortestLineIndex(lines, n);
	    _ = try stdout.print("{}\n", .{lines[shortest]});
	    lines[shortest] += 1;
	}
}

pub fn main() !void {
    var lines: [MAX_LINES]u32 = undefined;
    var n: usize = 0;
    var m: usize = 0;

    const stdin = std.io.getStdIn().reader();
    try stdin.readInt(&n, 10);
    try stdin.readInt(&m, 10);

    for (n) |i| {
	    try stdin.readInt(&lines[i], 10);
	}

    solve(&lines[0..n], n, m);
}
